### ISSUE TITLE
Investigate Trace / Context Interaction

### ISSUE THREAD

**Issue by bobstrecansky** [open-telemetry/opentelemetry-php#703]
We'd like to investigate [Trace/Context interaction](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#context-interaction)

We'd like to ensure we have good methods for getting and setting the active span.

**Comment by Grunet** [comment]
So if I'm following when the spec says "All the above functionalities operate solely on the context API, and they MAY be exposed as either static methods on the trace module, or as static methods on a class inside the trace module" does that recommend for PHP these be static methods, it's just up in the air which class exactly to stick them on?

I've also added some details on what a couple of other languages do below. Python, JS, and Java all appear to differ on which class got the relevant methods added to them.

@bobstrecansky @brettmc @tidal do you have opinions on where these methods should live?

TL;DR

This is what the APIs look like for Python I think - https://github.com/open-telemetry/opentelemetry-python/blob/540b8a5ed0fd79e77eced30eb514605bec97fe87/opentelemetry-api/src/opentelemetry/trace/propagation/__init__.py

So the usage is like as shown at https://opentelemetry.uptrace.dev/guide/python-tracing.html#context via the function from the trace module
```python
span = trace.get_current_span()
```

It's a little different in JS. It looks like the analogous helper method was removed at some point (this release? - https://github.com/open-telemetry/opentelemetry-js/blob/main/README.md#0140-to-0150) and so it now has to be
```js
api.getSpan(api.context.active())
api.setSpan(api.context.active(), span)
```

And Java brings back the helper method but I think is making it a static method if I'm understanding this correctly - https://opentelemetry.io/docs/instrumentation/java/manual/#get-the-current-span
```java
Span span = Span.current()
```

**Comment by Grunet** [comment]
I'm wondering if a trait would be an appropriate PHP-approach to this? I can try to draft one and see what it feels like

**Comment by Grunet** [comment]
I was thrown off by this language from the spec ""This is equivalent to getting the implicit context, then inserting the Span to the context." which seemed to imply the current context should be mutated, but after looking into what JS/Java/Python are doing I think the wording is just not ideal and you are supposed to make a new context like usual

JS - https://github.com/open-telemetry/opentelemetry-js-api/blob/bafcf0d0756e6ecc79fcc75f221b9dd17641eba5/src/context/context.ts#L46
Python - https://github.com/open-telemetry/opentelemetry-python/blob/d4d7c67663cc22615748d632e1c8c5799e8eacae/opentelemetry-api/src/opentelemetry/context/__init__.py#L99
Java - https://github.com/open-telemetry/opentelemetry-java/blob/main/context/src/main/java/io/opentelemetry/context/Context.java#L172

**Comment by Grunet** [comment]
Summarizing the discussion from the linked PR #720, I think functionality is in place to meet this part of the spec, but the exact presentation of the APIs might deserve a little tweaking, which can be looked at and evaluated in #723

Specifically the APIs for getting and setting the active span at this point in time are very similar to Java's, namely

```php
AbstractSpan::getCurrent()
$span->storeInContext(Context::getCurrent()) //Where $span's type extends the AbstractSpan abstract class 
``` 

If instrumentation libraries later need a shortcut to do the setter without an explicit reference to the Context class/instance, I think that should be possible to slot in as well (maybe without a BC depending on how #723 goes).

Given all that, and the fact that the spec compliance matrix looks up to date already, I think this issue can probably be closed (FYI @bobstrecansky )
